core/src/bundle/usBundleUtils.cpp

```diff
@@ -31,151 +30,103 @@
-#else
-  const bool sharedLibMode = false;
-#endif
+const char* dlerror(void)
+{
+  static std::string errStr;
+  errStr = us::GetLastErrorStr();
+  return errStr.c_str();
 }
 
-namespace BundleUtils
+void* dlopen(const char * path, int mode)
 {
+  (void)mode; // ignored
+  return reinterpret_cast<void*>(path == nullptr ? GetModuleHandle(nullptr) : LoadLibrary(path));
+}
 
-#ifdef __GNUC__
-
-std::string GetLibraryPath_impl(void* symbol)
+void* dlsym(void *handle, const char *symbol)
 {
-  Dl_info info;
-  if (dladdr(symbol, &info))
-  {
-    return info.dli_fname;
-  }
-  else
-  {
-    US_DEBUG << "GetLibraryPath_impl() failed for address " << symbol;
-  }
-  return "";
+  return reinterpret_cast<void*>(GetProcAddress(reinterpret_cast<HMODULE>(handle), symbol));
 }
 
-void* GetSymbol_impl(const std::string& bundleName, const std::string& libLocation, const char* symbol)
-{
-  // Clear the last error message
-  dlerror();
+#endif
 
-  void* selfHandle = nullptr;
-  if (!sharedLibMode || bundleName == "main")
-  {
-    // Get the handle of the executable
-    selfHandle = dlopen(0, RTLD_LAZY);
-  }
-  else
-  {
-    selfHandle = dlopen(libLocation.c_str(), RTLD_LAZY);
-  }
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 1024
+#endif
 
-  if (selfHandle)
-  {
-    void* addr = dlsym(selfHandle, symbol);
-    if (!addr)
-    {
-      const char* dlerrorMsg = dlerror();
-      if (dlerrorMsg)
-      {
-        US_DEBUG << "GetSymbol_impl() failed: " << dlerrorMsg;
-      }
-    }
-    dlclose(selfHandle);
-    return addr;
-  }
-  else
-  {
-    US_DEBUG << "GetSymbol_impl() dlopen() failed: " << dlerror();
-  }
-  return nullptr;
-}
+namespace us {
 
-#elif _WIN32
 
-std::string GetLibraryPath_impl(void *symbol)
+namespace BundleUtils
 {
-  HMODULE handle = nullptr;
-  BOOL handleError = GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
-                                       static_cast<LPCTSTR>(symbol), &handle);
-  if (!handleError)
-  {
-    // Test
-    US_DEBUG << "GetLibraryPath_impl():GetBundleHandle() " << GetLastErrorStr();
-    return "";
-  }
-
-  char bundlePath[512];
-  if (GetModuleFileName(handle, bundlePath, 512))
-  {
-    return bundlePath;
-  }
-
-  US_DEBUG << "GetLibraryPath_impl():GetBundleFileName() " << GetLastErrorStr();
-  return "";
+  
+void* GetExecutableHandle()
+{
+  return dlopen(0, RTLD_LAZY);;
 }
 
-void* GetSymbol_impl(const std::string& bundleName, const std::string& libLocation, const char* symbol)
+std::string GetExecutablePath()
 {
-  HMODULE handle = nullptr;
-  if (!sharedLibMode || bundleName == "main")
+  std::string execPath;
+  uint32_t bufsize = MAXPATHLEN;
+  std::unique_ptr<char[]> buf(new char[bufsize]);
+#if _WIN32
+  if (GetModuleFileName(nullptr, buf.get(), bufsize) == 0 || GetLastError() == ERROR_INSUFFICIENT_BUFFER)
   {
-    handle = GetModuleHandle(nullptr);
+    US_DEBUG << "GetModuleFileName failed" << GetLastErrorStr();
```

Comment:
@saschazelzer GetPrivate(GetBundleContext())->GetLogSink() does not work because BundleContextPrivate does not provide a GetLogSink method. It's in BundleContext class.